### PR TITLE
Move marker creation to the workspace

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Proposed Changes
Move the creation of the marker to the cursor class instead of the cursor_svg class.

### Reason for Changes
The marker is not solely a visual element and should therefore be available in a headless workspace.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/keyboard_nav/cursor.js [hunk 1]
```diff
@@ -28,15 +28,24 @@ goog.provide('Blockly.Cursor');
 
 /**
  * Class for a cursor.
+ * @param {Boolean} opt_isImmovable True if the cursor is a marker.
+ * False otherwise.
  * @constructor
  */
-Blockly.Cursor = function() {
+Blockly.Cursor = function(opt_isImmovable) {
```
Comment (RoboErikG) [inline core/keyboard_nav/cursor.js]: For consistency can this be opt_movable and default to true?

File: core/keyboard_nav/cursor.js [hunk 2]
```diff
@@ -28,15 +28,24 @@ goog.provide('Blockly.Cursor');
...
   /*
    * The current location of the cursor.
    * @type {Blockly.Field|Blockly.Connection|Blockly.Block}
    * @private
    */
   this.curNode_ = null;
+
+  /**
+   * Whether or not the cursor is a marker.
+   * @type {Boolean} True if the cursor is a marker. False otherwise.
```
Comment (samelhusseini) [inline core/keyboard_nav/cursor.js]: lowercase boolean.

File: core/keyboard_nav/cursor.js [hunk 3]
```diff
@@ -28,11 +28,11 @@ goog.provide('Blockly.Cursor');
 
 /**
  * Class for a cursor.
- * @param {boolean} opt_movable False if the cursor is a marker. True or
- * undefined if the cursor is not a marker.
+ * @param {boolean} opt_marker False if the cursor is a marker. True or
```
Comment (RoboErikG) [inline core/keyboard_nav/cursor.js]: flip this. If marker === true it's a marker.

File: core/keyboard_nav/cursor.js [hunk 4]
```diff
@@ -28,15 +28,24 @@ goog.provide('Blockly.Cursor');
 
 /**
  * Class for a cursor.
+ * @param {boolean} opt_marker True if the cursor is a marker. False or
```
Comment (NeilFraser) [inline core/keyboard_nav/cursor.js]: This is an optional argument.  {boolean=}

**Pull request merged**